miden-lib/asm/sat/utils.masm

```diff
@@ -30,6 +59,69 @@ end
 #!
 #! - i is the index of the consumed note.
 #! - ptr is the nullifier memory address for note i.
-export.consumed_note_nullifier_ptr
+export.get_consumed_note_nullifier_ptr
     push.CONSUMED_NOTE_SECTION_OFFSET.1 add add
 end
+
+#! Returns the initial account hash.
+#!
+#! Stack: []
+#! Output: [INIT_ACCT_HASH]
+#!
+#! - INIT_ACCT_HASH is the initial account hash.
+export.get_initial_account_hash
+    padw push.INIT_ACCT_HASH_PTR mem_loadw
+end
+
+#! Computes and returns the account hash from account data stored in memory.
+#!
+#! Stack: []
+#! Output: [ACCT_HASH]
+#!
+#! - ACCT_HASH is the hash of the account data.
+export.get_account_hash
+    # prepare the stack for computing the account hash
+    push.ACCT_DATA_SECTION_OFFSET padw padw padw
+
+    # stream account data and compute sequential hash
+    mem_stream mem_stream
+
+    # extract account hash
+    dropw swapw dropw
+
+    # drop memory pointer
+    movup.4 drop
+end
+
+#! Returns the initial account nonce.
+#!
+#! Stack: []
+#! Output: [INIT_ACCT_NONCE]
+#!
+#! - INIT_ACCT_NONCE is the initial account nonce.
+export.get_initial_account_nonce
+    push.INIT_NONCE_PTR mem_load
+end
+
+#! Returns the account nonce.
+#!
+#! Stack: []
+#! Output: [acct_nonce]
+#!
+#! - acct_nonce is the account nonce.
+export.get_account_nonce
+    padw push.ACCT_ID_AND_NONCE_PTR mem_loadw
+    movdn.3 drop drop drop
+end
+
+#! Increments the account nonce by the provided value.
+#!
+#! Stack: [value]
+#! Output: []
+#!
+#! - value is the value to increment the nonce by.
+export.incr_account_nonce
+    padw push.ACCT_ID_AND_NONCE_PTR mem_loadw
+    movup.4 u32checked_add
+    push.ACCT_ID_AND_NONCE_PTR mem_storew dropw
+end
```

Comment:
I was thinking we'd check only the input value (not the nonce itself). So, it would look like:
```
export.incr_account_nonce
    u32assert.1
    padw push.ACCT_ID_AND_NONCE_PTR mem_loadw
    movup.4 add
    push.ACCT_ID_AND_NONCE_PTR mem_storew dropw
end
```
I'm not sure we should restrict the nonce to be 32 bits.

Reply:
got it, makes sense!